data/adapters/SEM_motif_adapter.py

```diff
@@ -0,0 +1,134 @@
+
+
+class SEMMotif(Adapter):
+    ALLOWED_LABELS = ['motif', 'motif_protein_link']
+    SOURCE = 'SEMpl'
+    SOURCE_URL = 'https://github.com/Boyle-Lab/SEMpl'
+    TF_PROTEIN_MAPPING_PATH = './data_loading_support_files/SEMVAR_provenance_uniprot_ids.csv'
+
+    OUTPUT_PATH = './parsed-data'
+
+    def __init__(self, filepath, label='motif', dry_run=True):
+        if label not in SEMMotif.ALLOWED_LABELS:
+            raise ValueError('Ivalid label. Allowed values: ' +
+                             ','.join(SEMMotif.ALLOWED_LABELS))
+
+        self.filepath = filepath
+        self.label = label
+        self.dataset = label
+        if label == 'motif':
+            self.type = 'node'
+            self.collection = 'motifs'
+        else:
+            self.type = 'edge'
+            self.collection = 'motifs_proteins'
+
+        self.dry_run = dry_run
+        self.output_filepath = '{}/{}_{}.json'.format(
+            self.OUTPUT_PATH,
+            self.SOURCE,
+            self.dataset
+        )
+
+        super(SEMMotif, self).__init__()
+
+    def load_tf_id_mapping(self):
+        self.tf_id_mapping = {}
+        with open(SEMMotif.TF_PROTEIN_MAPPING_PATH, 'r') as map_file:
+            map_csv = csv.reader(map_file)
+            for row in map_csv:
+                if row[2]:  # this is a complex
+                    self.tf_id_mapping[row[0]] = 'complexes/' + row[2]
+                else:
+                    self.tf_id_mapping[row[0]] = 'proteins/' + row[3]
+
+    def process_file(self):
+        parsed_data_file = open(self.output_filepath, 'w')
+        for filename in os.listdir(self.filepath):
+            if filename.endswith('.sem'):
+                motif_id = filename.split('.')[0]  # e.g. M00778
+                if self.label == 'motif':
+                    pwm = []
+                    with open(self.filepath + '/' + filename, 'r') as sem_file:
+                        # read tf name from the first line of sem file
+                        tf_name = next(sem_file).strip().split()[0]
+                        for row in sem_file:
+                            sem_row = row.strip().split()[1:]
+                            pwm.append([str(value) for value in sem_row])
+
+                    length = len(pwm)
+                    _key = tf_name + '_' + SEMMotif.SOURCE
+                    props = {
+                        '_key': _key,
+                        'name': _key,
+                        'tf_name': tf_name,
+                        'source': SEMMotif.SOURCE,
+                        'source_url': SEMMotif.SOURCE_URL,
+                        'pwm': pwm,
+                        'length': length
+                    }
+                elif self.label == 'motif_protein_link':
+                    self.load_tf_id_mapping()
+                    with open(self.filepath + '/' + filename, 'r') as sem_file:
```

Comment:
Those two lines are duplicated, can you refactor?

Reply:
yes refactored.